contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+contract DomainRegistry {
+    uint256 constant REQUIRED_DEPOSIT = 1 ether;
+
+    struct Domain {
+        uint256 deposit;
+        bool isRegistered;
+    }
+
+    mapping(string => Domain) public domains;
+
+    event DomainCreated(string tld, uint256 deposit);
+    event DomainReleased(string tld);
```

Comment:
What was you approach in choosing which fields to include in the events? Which of them to index or not?
How can it help with data monitoring?

Reply:
fxd